dartagnan/src/main/java/com/dat3m/dartagnan/encoding/EncodingContext.java

```diff
@@ -120,6 +121,10 @@ public BooleanFormula encodeBooleanExpressionAt(ExprInterface expression, Event
     }
 
     public Formula encodeIntegerExpressionAt(ExprInterface expression, Event event) {
+        if (expression instanceof VirtualMemoryObject) {
+            return new ExpressionEncoder(formulaManager, event).encodeAsInteger(
+                    ((VirtualMemoryObject) expression).getPhysicalAddress());
+        }
```

Comment:
I think you should move this code into the `ExpressionEncoder`.
It has a `visit(MemoryObject)` method that could do the check and extract the physical address if necessary.

Reply:
This code seems not needed for the new physical memory implementation, think I'll just remove it.